pkg/api/targetgroup.go

```diff
@@ -11,45 +10,16 @@ import (
 	"github.com/common-fate/ddb"
 )
 
-func (a *API) FetchTargetGroups(ctx context.Context) []types.TargetGroup {
```

Comment:
@meyerjrr Our service layer should be responsible for business logic,
OUr API layer is responsible for data fetching, serialising and calling service layer methods

In this PR you have shifted data fetching code from the API to the service layer, this goes against our desired separation of concerns here and abstracts the data fetching from the API

The way that this code handles no items returned and the fact that it packages them into an API type is not a concern of the service layer.

For this PR you should revert removing this fetch resources function and instead keep data fetching as an operation of the API layer